src/internal.c

```diff
@@ -28063,17 +28166,33 @@ static int DoServerKeyExchange(WOLFSSL* ssl, const byte* input,
                     #ifdef HAVE_ECC
                         case ecc_dsa_sa_algo:
                         {
-                            ret = EccVerify(ssl,
-                                args->verifySig, args->verifySigSz,
-                                ssl->buffers.digest.buffer,
-                                ssl->buffers.digest.length,
-                                ssl->peerEccDsaKey,
-                            #ifdef HAVE_PK_CALLBACKS
-                                &ssl->buffers.peerEccDsaKey
-                            #else
-                                NULL
-                            #endif
-                            );
+                        #ifdef HAVE_PK_CALLBACKS
+                            ret = NOT_COMPILED_IN;
+                            if (ssl->options.side == WOLFSSL_CLIENT_END &&
+                                ssl->ctx && ssl->ctx->ProcessServerKexCb) {
+                                ret = ssl->ctx->ProcessServerKexCb(ssl,
+                                    args->sigAlgo,
+                                    args->verifySig, args->verifySigSz,
+                                    ssl->buffers.sig.buffer, SEED_LEN,
+                                    &ssl->buffers.sig.buffer[SEED_LEN],
+                                    (ssl->buffers.sig.length - SEED_LEN));
+                            }
+
+                            if (ret == NOT_COMPILED_IN)
+                        #endif /* HAVE_PK_CALLBACKS */
+                            {
+                                ret = EccVerify(ssl,
```

Comment:
TODO: This requires a bit more testing and investigation. Its saying "Prepare for server key exchange (keygen?) and if successful, skip over ECDSA verification. Very strange.

Reply:
Created `EccSkipSharedSecretCb` to accommodate this.  The callback should be renamed to indicate both signature  verification and shared secret stuff is happening here.

Reply:
Fixed in 57ef02d6f2ea94710ac19d854756a1d95e75464d